main/javascript-modifications/ses-reference.md

```diff
@@ -0,0 +1,286 @@
+# SES Reference
+
+This document describes how SES (Secure ECMAScript) affects writing Agoric JavaScript code. 
+It is very much a "how to do something" document, with little explanation about why and
+how something was implemented or other background information. For that, see the more
+comprehensive [SES Guide](./ses-guide.md).
+
+## Installing SES in Your Code
+
+## Removed by SES summary
+
+The following are missing or unusable under SES:
+- Most [Node.js-specific global objects](https://nodejs.org/dist/latest-v14.x/docs/api/globals.html) 
+- All [Node.js built-in modules](https://nodejs.org/dist/latest-v14.x/docs/api/) such as `http` and 
+  `crypto`. 
+- [Features from browser environments](https://developer.mozilla.org/en-US/docs/Web/API) presented as names in the global scope including `atob`, `TextEncoder`, and `URL`.
+- HTML comments
+- Dynamic `import` expressions
+- Direct evals
+
+## Added/Changed by SES summary
+
+SES adds the following to JavaScript or changes them significantly: 
+- console	
+- `Compartment`
+- `lockdown()`
+- `harden()`
+- `globalThis` is frozen.
+- JavaScript primordials are frozen.
+
+## `lockdown()`
+
+`lockdown()` tamper-proofs all of the JavaScript intrinsics, so no program can subvert their methods
+(preventing some man in the middle attacks). Also, no program can use them to pass notes to parties
+that haven't been expressly introduced (preventing some covert communication channels).
+
+Lockdown *freezes* all JavaScript defined objects accessible to any program in the realm. The frozen 
+accessible objects include but are not limited to: **tyg todo: Is there a short English description for
+the 2nd-4th items here?**
```

Comment:
- `[].__proto__` is the array prototype, equivalent to `Array.prototype` in a pristine JavaScript environment.
- `{}.__proto__` is `Object.prototype` 〃
- `(() => {}).__proto__` is `Function.prototype` 〃
- `(async () => {}).__proto__` is the prototype of all asynchronous functions and has no alias in the global scope of a pristine JavaScript environment.